src/com/owncloud/android/services/observer/InstantUploadsObserverBasedOnINotify.java

```diff
@@ -0,0 +1,194 @@
+        }
+
+        mConfiguration = configuration;
+        mContext = context;
+        mObservedChildren = new HashMap<>();
+        mInstantUploadsHandler = new InstantUploadsHandler();
+    }
+
+    /**
+     * Receives and processes events about updates of the monitored folder.
+     *
+     * This is almost heuristic. Do no expect it works magically with any camera.
+     *
+     * For instance, Google Camera creates a new video file when the user enters in "video mode", before
+     * start to record, and saves it empty if the user leaves recording nothing. True store. Life is magic.
+     *
+     * @param event     Kind of event occurred.
+     * @param path      Relative path of the file referred by the event.
+     */
+    @Override
+    public void onEvent(int event, String path) {
+        Log_OC.d(TAG, "Got event " + event + " about "  + ((path != null) ? path : "") +
+            " in current camera folder");
+
+        if (path != null && path.length() > 0) {
+            synchronized (mLock) {
+                Log_OC.d(TAG, "Observed camera folder is " + mConfiguration.getSourcePath());
+
+                if ((event & FileObserver.CREATE) != 0) {
+                    // new file created, let's watch it; false -> not modified yet
+                    mObservedChildren.put(path, false);
+                }
+                if ((   (event & FileObserver.MODIFY) != 0) &&
+                        mObservedChildren.containsKey(path) &&
+                        !mObservedChildren.get(path)
+                    ) {
+                    // watched file was written for the first time after creation
+                    mObservedChildren.put(path, true);
+                }
+                if (   (event & FileObserver.CLOSE_WRITE) != 0 &&
+                        mObservedChildren.containsKey(path)    &&
+                        mObservedChildren.get(path)
+                    ) {
+                    // a file that was previously created and written has been closed;
+                    // testing for FileObserver.MODIFY is needed because some apps
+                    // close the video file right after creating it when the recording
+                    // is started, and reopen it to write with the first chunk of video
+                    // to save; for instance, Camera MX does so.
+                    mObservedChildren.remove(path);
+                    handleNewFile(path);
+                }
+                if ((event & FileObserver.MOVED_TO) != 0) {
+                    // a file has been moved or renamed into the folder;
+                    // for instance, Google Camera does so right after
+                    // saving a video recording
+                    handleNewFile(path);
+                }
+            }
+        }
+
+        if ((event & IN_IGNORE) != 0 &&
+            (path == null || path.length() == 0)) {
+            Log_OC.d(TAG, "Stopping the observance on " + mConfiguration.getSourcePath());
```

Comment:
The event is received *after* the stop occurs, not before. The system decided to stop sending events about changes in the file, or our observer was unsuscribed of the events before.